### PR TITLE
3736 - implemented the endpoint for the provider

### PR DESCRIPTION
**Pull request by avinhurry**
### Changes proposed in this pull request

Added documentation for the null provider 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by avinhurry** [comment]
[[S] Implement API endpoint - /api/public/v1/recruitment_cycles/:year/providers/:provider_code](https://trello.com/c/zI3kyVcI/3736-s-implement-api-endpoint-api-public-v1-recruitmentcycles-year-providers-providercode)

**Comment by asmega** [comment]
Just a note your `include` won't work because theres is only one serializer specified, we will have to swap this out once we implement more. It also won't work because haven't specified `belongs_to` on the provider serializer

File: app/controllers/api/public/v1/providers_controller.rb [hunk 1]
```diff
@@ -13,19 +13,16 @@ def index
...
+          code = params.fetch(:code, params[:provider_code])
+          provider = recruitment_cycle.providers
+                                        .find_by!(
+                                          provider_code: code.upcase,
+                                        )
+
+          render jsonapi: provider,
+                 class: { Provider: API::Public::V1::SerializableProvider },
+                 include: params[:include],
+                 fields: { providers: provider_fields }
```
Comment (asmega) [inline app/controllers/api/public/v1/providers_controller.rb:25]: not something made in this PR but `provider_fields` should return nil if nothing is specified. The default behaviour seems to serialize everything defined in the serializer if it is nil, therefore there is no need to have a default provider fields passed in.

File: app/controllers/api/public/v1/application_controller.rb [hunk 2]
```diff
@@ -2,6 +2,11 @@ module API
   module Public
     module V1
       class ApplicationController < ActionController::API
+        rescue_from ActiveRecord::RecordNotFound, with: :jsonapi_404
```
Comment (darokel) [inline app/controllers/api/public/v1/application_controller.rb:5]: I think we need a standardised way of dealing with these types of errors. What about something like:

```
    def render_json_error(code: nil, message:, status: :internal_server_error)
      render json: { code: code, error: message }.compact, status: status
    end
```

File: app/controllers/api/public/v1/providers_controller.rb (hunk 1 above)
Comment (avinhurry) [reply app/controllers/api/public/v1/providers_controller.rb]: Done, thanks.

File: app/controllers/api/public/v1/application_controller.rb (hunk 2 above)
Comment (avinhurry) [reply app/controllers/api/public/v1/application_controller.rb:5]: I've added a card for this to my ideas bored. We can chat about it

**Pull request merged**